com.bladecoder.adventure-editor.json

```diff
@@ -0,0 +1,84 @@
+{
+    "app-id" : "com.bladecoder.adventure-editor",
+    "runtime" : "org.freedesktop.Platform",
+    "runtime-version" : "1.6",
+    "sdk" : "org.freedesktop.Sdk",
+    "sdk-extensions" : [
+        "org.freedesktop.Sdk.Extension.openjdk9"
+    ],
+    "command" : "adventure-editor",
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--socket=pulseaudio",
+        "--device=dri",
+        "--filesystem=home",
+         "--env=PATH=/usr/bin:/app/bin:/app/jre/bin",
+         "--env=JAVA_HOME=/app/jre"
+    ],
+    "modules" : [
+        {
+            "name": "xorg-xrandr",
+            "buildsystem": "simple",
```

Comment:
This should be the regular build system, you aren't doing anything advanced on top. You can make an `autogen.sh` with the `script` type source file that calls `autoreconf -fi` if needed.

Reply:
Sorry, I don't understand what should I do here. Can you point me to other project that do this or to the doc to view how to change this?

Reply:
Like this: https://github.com/flathub/org.DolphinEmu.dolphin-emu/blob/98d37a870252a19e8e31e0b3841eb24788904526/org.DolphinEmu.dolphin-emu.json#L49-L55

Reply:
I'm trying adding this to the xorg sources, but doesn't work for me:

```
                {
                     "type": "script",
                     "dest-filename": "autogen.sh",
                     "commands": [
                          "autoreconf -fi"
                     ]
                }
```

I think that the autogen.sh isn't called. If I use type `shell` instead of `script` it works.